packages/frontend/src/lib/pnl.ts

```diff
@@ -70,25 +82,47 @@ export async function calcETHCollateralPnl(
           )
         }
         return acc
-      }, Promise.resolve({ deposits: BIG_ZERO, withdrawals: BIG_ZERO }))
-    : { deposits: BIG_ZERO, withdrawals: BIG_ZERO }
+      }, Promise.resolve({ deposits: BIG_ZERO, withdrawals: BIG_ZERO, priceError: false }))
+    : { deposits: BIG_ZERO, withdrawals: BIG_ZERO, priceError: true }
 
-  return currentVaultEthBalance.times(uniswapEthPrice).minus(deposits.minus(withdrawals))
+  return !priceError ? currentVaultEthBalance.times(uniswapEthPrice).minus(deposits.minus(withdrawals)) : BIG_ZERO
+}
+
+const getRelevantSwaps = (squeethAmount: BigNumber, swaps: swaps_swaps[], isWethToken0: boolean, isLong = false) => {
+  let totalSqueeth = BIG_ZERO
+  const relevantSwaps = []
+  for (let index = swaps.length - 1; index >= 0; index--) {
+    const squeethAmt = new BigNumber(isWethToken0 ? swaps[index].amount1 : swaps[index].amount0)
+    const sqthAmount = isLong ? squeethAmt.negated() : squeethAmt
+    totalSqueeth = totalSqueeth.plus(sqthAmount)
+    relevantSwaps.push(swaps[index])
+    if (squeethAmount.isEqualTo(totalSqueeth)) {
```

Comment:
Just wondering if this really works cuz if we going from the end of swaps data, the summation of users' position might go wrong

Reply:
It's asc so the most recent data are at the end.

Reply:
Let's say user has long 2 osqth first, and then mint 2 osqth and sell 2 osqth, from the end of swaps, you will get -2(sell), +2(long) = 0, but the actual position will be 2 long position and 2 short position, so if we consider direct selling, which we already did rn, the calculation of actual long & short will not be just accumulated all txs up, so prob we cannot take swaps data like this, and Im still working on to get correct direct selling short position 

This will work for most users who did not do direct selling on uniswap, so not sure if we want to have this temporary fix here

Reply:
Ok so @yichiehliu you'd fix this in the next PR ?

Reply:
yh so I think can merge this one first and then update the pnl calcs then